src/main/java/ay2122s1_cs2103t_w16_2/btbb/storage/JsonAdaptedOrder.java

```diff
@@ -72,6 +105,56 @@ public Order toModelType() throws IllegalValueException {
         }
         final Address modelClientAddress = new Address(clientAddress);
 
-        return new Order(modelClientName, modelClientPhone, modelClientAddress);
+        if (recipeName == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, GenericString.getMessageConstraints("Recipe Name")
+            ));
+        }
+        if (!GenericString.isValidGenericString(recipeName)) {
+            throw new IllegalValueException(GenericString.getMessageConstraints("Recipe Name"));
+        }
+        final GenericString modelRecipeName = new GenericString(recipeName);
+
+        final List<Ingredient> ingredients = new ArrayList<>();
+        for (JsonAdaptedIngredient ingredient : recipeIngredients) {
+            Ingredient i = ingredient.toModelType();
+            boolean hasValidQuantity = GenericQuantity.isValidQuantity(i.getQuantity().toString());
+
+            if (hasValidQuantity) {
+                ingredients.add(i);
+            }
+        }
+        final RecipeIngredientList modelRecipeIngredients = new RecipeIngredientList(ingredients);
+
+        if (price == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Price.class.getSimpleName()));
+        }
+        if (!Price.isValidPrice(price)) {
+            throw new IllegalValueException(Price.MESSAGE_CONSTRAINTS);
+        }
+        final Price modelPrice = new Price(price);
+
+        if (deadline == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, Deadline.class.getSimpleName()
+            ));
+        }
+        if (!Deadline.isValidInternalDeadline(deadline)) {
+            throw new IllegalValueException(Deadline.MESSAGE_CONSTRAINTS);
+        }
+        final Deadline modelDeadline = new Deadline(deadline);
+
+        if (quantity == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, GenericQuantity.class.getSimpleName()
+            ));
```

Comment:
If you're using GenericQuantity now, is this supposed to change too?

Reply:
This comment is no longer valid after reverting the changes made to the lastest commit